controllers/controlplaneupgrade_controller.go

```diff
@@ -211,11 +232,19 @@ func (r *ControlPlaneUpgradeReconciler) updateStatus(ctx context.Context, log lo
 	log.Info("Updating ControlPlaneUpgrade status")
 	nodeUpgrade := &anywherev1.NodeUpgrade{}
 	nodesUpgradeCompleted := 0
-	nodesUpgradeRequired := len(cpUpgrade.Spec.MachinesRequireUpgrade)
-	for _, machineRef := range cpUpgrade.Spec.MachinesRequireUpgrade {
-		if err := r.client.Get(ctx, GetNamespacedNameType(nodeUpgraderName(machineRef.Name), constants.EksaSystemNamespace), nodeUpgrade); err != nil {
-			return fmt.Errorf("getting node upgrader for machine %s: %v", machineRef.Name, err)
+	nodesUpgradeRequired := len(machines)
+	machineStates := []anywherev1.MachineState{}
+
+	for _, machine := range machines {
```

Comment:
Just a thought, should we append each machine we upgrade to the status in the reconcile method, and here only update the status based on what is already in the list instead of all of them? It decreases the number of calls, especially because not all the machines will show up in the status anyways. If we keep this approach, I would consider trying to find a way to just include all the machines in the status, and maybe adding a field for which machine is the first control plane or something.